expr/interval_compound_literal.go

```diff
@@ -0,0 +1,154 @@
+package expr
+
+import (
+	"errors"
+
+	"github.com/substrait-io/substrait-go/proto"
+	"github.com/substrait-io/substrait-go/types"
+)
+
+type intervalDateParts struct {
+	years      int32
+	months     int32
+	days       int32
+	seconds    int32
+	subSeconds int64
+}
+
+type intervalDatePartsOptions func(parts *intervalDateParts)
+
+func WithIntervalCompoundYears(years int32) func(*intervalDateParts) {
```

Comment:
Let's just hang these methods off of the literal object. So if I wanted, I could do:

new IntervalCompoundLiteral{}.WithMillis(1).WithSeconds(2)...etc.

Reply:
fixed it